tests/nightly/test_all.sh

```diff
@@ -72,9 +80,15 @@ check_val() {
 example_dir=../../example/image-classification
 # python: lenet + mnist
 test_lenet() {
-    python $example_dir/train_mnist.py \
-        --data-dir `pwd`/data/mnist/ --network lenet --gpus $gpus --num-epochs 10 \
-        2>&1 | tee log
+    optimizer=$(get_random_optimizer)
```

Comment:
My concern was that it might be hard to reproduce failures. Granted that someone who knows the test and looks at the specific failure should be able to figure it out. For example, the bug you fixed yesterday, would that have been immediately caught by this test?

I do see your concern about longer run time.